flash/vision/classification/data.py

```diff
@@ -13,369 +13,392 @@
 
-        if isinstance(samples, str):
-            samples = [samples]
-        if isinstance(samples, (list, tuple)) and all(isinstance(p, str) for p in samples):
-            outputs = []
-            for sample in samples:
-                try:
-                    output = self._loader(sample)
-                    transform = self._valid_transform if self._use_valid_transform else self._train_transform
-                    outputs.append(transform(output))
-                except UnidentifiedImageError:
-                    print(f'Skipping: could not read file {sample}')
+    @staticmethod
+    def load_sample(sample) -> Union[Image.Image, list]:
+        # open path as file to avoid ResourceWarning (https://github.com/python-pillow/Pillow/issues/835)
+        if isinstance(sample, torch.Tensor):
+            return sample
+
+        if isinstance(sample, (tuple, list)):
+            path = sample[0]
+            sample = list(sample)
+        else:
+            path = sample
 
-            return outputs
-        raise MisconfigurationException("The samples should either be a tensor or a list of paths.")
+        with open(path, "rb") as f, Image.open(f) as img:
+            img = img.convert("RGB")
+
+        if isinstance(sample, list):
+            sample[0] = img
+            return sample
+
+        return img
+
+    @classmethod
+    def predict_load_data(cls, samples: Any) -> Any:
+        if isinstance(samples, torch.Tensor):
+            return samples
+        return cls._get_predicting_files(samples)
+
+    def _convert_tensor_to_pil(self, sample):
+        #  some datasets provide their data as tensors.
+        # however, it would be better to convert those data once in load_data
+        if isinstance(sample, torch.Tensor):
+            sample = to_pil_image(sample)
+        return sample
+
+    def _apply_transform(
+        self, sample: Any, transform: Union[Callable, Dict[str, Callable]], func_name: str
+    ) -> torch.Tensor:
+        if transform is not None:
+            if isinstance(transform, (Dict, ModuleDict)):
+                if func_name not in transform:
+                    return sample
+                else:
+                    transform = transform[func_name]
+            sample = transform(sample)
+        return sample
+
+    def collate(self, samples: Sequence) -> Any:
+        _samples = []
+        # todo: Kornia transforms add batch dimension which need to be removed
+        for sample in samples:
+            if isinstance(sample, tuple):
+                sample = (sample[0].squeeze(0), ) + sample[1:]
+            else:
+                sample = sample.squeeze(0)
+            _samples.append(sample)
+        return default_collate(_samples)
+
+    def common_pre_tensor_transform(self, sample: Any, transform) -> Any:
+        return self._apply_transform(sample, transform, "pre_tensor_transform")
+
+    def train_pre_tensor_transform(self, sample: Any) -> Any:
+        sample, target = sample
```

Comment:
```suggestion
        source, target = sample
```
having sample also after split is coonfusing

Reply:
Done !